compiler/parser.nim

```diff
@@ -100,6 +103,9 @@ template prettySection(body) =
 proc getTok(p: var Parser) =
   ## Get the next token from the parser's lexer, and store it in the parser's
   ## `tok` member.
+  p.lineNumberPrevious = p.lex.lineNumber
+  p.lineStartPrevious = p.lex.lineStart
+  p.bufposPrevious = p.lex.bufpos
```

Comment:
Can't this be done off the critical path, only for the sections of the parser where it's used later on?

Reply:
the cost of `getTok` is dominated by `rawGetTok, those 3 lines added are not significant (and even then, `getTok` is not a hotspot).
Also, I tried with and without those 3 lines, it doesn't change compilation performance on compiling a program such as compiler/nim.nim itself.

Complicating `getTok` is not worth it, those fields can be reused in other contexts.